### DISCUSSION TITLE
issue with Flow of code

### DISCUSSION THREAD

**Discussion by Arjun5421** [Cyfrin/foundry-full-course-cu#1600] [General]
please explain the working of retrieve function in storage factory (lesson 3).

**Comment by EngrPips** [comment]
Hello @Arjun5421 , the `retrieve` function basically help you get whatever you have initially stored in the contract as your `favorite number`

**Reply by Arjun5421** [reply]
the retrieve function gets the variable by the name of myfavouritenumber as defined in simple storage. So how is it able to get _simpleStorageNumber as myfavouritenumber and _simpleStorageNumber are not same

**Reply by xueyuanhuang** [reply]
Hello @Arjun5421 , I am not sure what you're trying to express, you may ask in more detail.
According to your sentence:`So how is it able to get _simpleStorageNumber as myfavouritenumber and _simpleStorageNumber are not same`

I would guess you refer to `StorageFactory.sol
`, then it just upgrade version of `SimpleStorage.sol`.

Basically, it means that `StorageFactory.sol` can have lots of contracts. And if the contracts are different, it make sense that the ` myfavouritenumber` are different, as they are different contracts.

**Reply by EngrPips** [reply]
@xueyuanhuang  thanks for the input. @Arjun5421  do you understand now ?

**Reply by xueyuanhuang** [reply]
Maybe it also works to not thinking too much, just put aside the confusion.

Keep learning first, later it will really lots easier after you have finished other part haha.

Or try to use chatgpt😇😇😇😇
